cmake/modules/AddSwiftHostLibrary.cmake

```diff
@@ -97,6 +97,16 @@ function(add_swift_syntax_library name)
     $<$<COMPILE_LANGUAGE:Swift>:-color-diagnostics>
   )
 
+  if(BOOTSTRAPPING_MODE STREQUAL "CROSSCOMPILE")
```

Comment:
As I've mentioned in the other PRs - the goal is to require a builder toolchain, fully remove `BOOTSTRAPPING_MODE`, and to properly run multiple CMake invocations. We are moving towards that goal, but adding more references bootstrapping everywhere is opposite to the direction we want to go.

Reply:
You're going to have to clarify what you mean here, as I did on that other pull of yours.

> fully remove BOOTSTRAPPING_MODE

You said earlier that you want to remove [two of the options for `BOOTSTRAPPING_MODE`, `BOOTSTRAPPING` and  `BOOTSTRAPPING-WITH-HOSTLIBS`](https://github.com/apple/swift/blob/1ce3935eb77ab564fb700c09aed4ca45d60c854b/CMakeLists.txt#L346). When I responded that that was fine but that we'd still need to differentiate between the other three options listed there, [you didn't respond](https://github.com/apple/swift/pull/70343#discussion_r1483920819).

> properly run multiple CMake invocations

No idea what this refers to, as building a cross-compiled toolchain like this already involves dozens of CMake invocations, a couple times for each Swift repo.

> adding more references bootstrapping everywhere is opposite to the direction we want to go.

This is a single reference, because this is the build variable that currently exists to do it. Whenever you replace this build variable in the future, we can easily update this to use your new scheme.